### ISSUE TITLE
Support Configurable Session Timeouts Per Target

### ISSUE THREAD

**Issue by wolveix** [warp-tech/warpgate#429]
Hey guys. Loving using Warpgate, but I frequently leave a connection open to a target for extended periods of time (20 minutes - 60 minutes), and have noticed that Warpgate closes the connection much sooner. Would it be possible to make this timeout configurable?

**Comment by erd05** [comment]
I would also love this. The connection close after 60 seconds which is really too small. I can't seem to find the variable inside the code to change it globally even.

**Comment by Eugeny** [comment]
What SSH clients are you using? SSH times out on the server side and this is normally circumvented by keepalive messages. Warpgate doesn't send its own, but will forward keepalive messages if the client sends any.

**Comment by erd05** [comment]
OpenSSH_8.4p1 Debian-5+deb11u1, OpenSSL 1.1.1n  15 Mar 2022

**Comment by erd05** [comment]
Added:
`Host *
    ServerAliveInterval 30
    ServerAliveCountMax 2`

To my ~/.ssh/config (30 sec between keepalive message and 2 messages max before declaring dead) and now it's working great thank you!

**Comment by notnooblord** [comment]
I also tried setting server side parameter
```
        ClientAliveInterval 45
        ClientAliveCountMax 10
```

But seems like warpgate do not forward these keepalives to client, so those have no effect.
Maybe would be good to add some option that is similar to ssh server, that never drops tcp socket?
Even if there is no data on wire.

**Comment by markusglaetzner** [comment]
I use warpgate together with xpipe and I only can set configs on my jump-servers (warpgate runs on my jump-servers) and the target servers, but with no effect. I would likt to set the above parameters in "~/.ssh/config" on the jump-servers, but the Account is the one from warpgate (admin) and there is no userhome for this account. It would be helpfull, if the parameter could be set in the web-portal of warpgate.

**Comment by wolveix** [comment]
Seems this was closed by #990 :)
